helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -336,7 +342,32 @@ public Integer apply(final Response reply) {
   }
 
   public ListenableFuture<List<String>> listHosts() {
-    return get(uri("/hosts/"), new TypeReference<List<String>>() {
+    return listHosts(ImmutableMultimap.<String, String>of());
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern) {
+    return listHosts(ImmutableMultimap.of("namePattern", namePattern));
+  }
+
+  public ListenableFuture<List<String>> listHosts(final List<String> labels) {
+    final Multimap<String, String> query = HashMultimap.create();
+    query.putAll("labels", labels);
+
+    return listHosts(query);
+  }
+
+  public ListenableFuture<List<String>> listHosts(final String namePattern,
+                                                  final List<String> labels) {
```

Comment:
Although a query parameter like `%7B%22label%22%3A%22foo%22%2C%22operator%22%2C%22EQUALS%22%2C%22operand%22%2C%22bar%22%7D` isn't terrible unless a human is supposed to craft it :)